sdks/python/apache_beam/io/filebasedsink_test.py

```diff
@@ -269,28 +276,110 @@ def test_file_sink_multi_shards(self):
     # Check that any temp files are deleted.
     self.assertItemsEqual(res, glob.glob(temp_path + '*'))
 
-  def test_file_sink_io_error(self):
-    temp_path = os.path.join(self._new_tempdir(), 'ioerror')
+  @mock.patch.object(filebasedsink.FileSystems, 'rename')
+  def test_file_sink_rename_error(self, filesystems_mock):
+    temp_path = os.path.join(self._new_tempdir(), 'rename_error')
+    sink = MyFileBasedSink(
+        temp_path, file_name_suffix='.output', coder=coders.ToStringCoder())
+    init_token, writer_results = self._common_init(sink)
+    pre_finalize_results = sink.pre_finalize(init_token, writer_results)
+
+    error_str = 'mock rename error description'
+    filesystems_mock.side_effect = BeamIOError(
+        'mock rename error', {('src', 'dst'): error_str})
+    with self.assertRaisesRegexp(Exception, error_str):
+      list(sink.finalize_write(init_token, writer_results,
+                               pre_finalize_results))
+
+  def test_file_sink_src_missing(self):
```

Comment:
<!--new_thread; commit:24c81c1018b47c5316f09f3b1eef8725b5492848; resolved:0-->
test_file_sink_src_and_dest_missing_fails

Reply:
In this case the rename should succeed, except that rename() has been stubbed out to return an error.